ror-tools-core/src/main/scala/tech/beshu/ror/tools/core/patches/base/EsPatchExecutor.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.tools.core.patches.base
+
+import tech.beshu.ror.tools.core.patches.base.EsPatchExecutor.EsPatchStatus
+import tech.beshu.ror.tools.core.patches.internal.RorPluginDirectory
+import tech.beshu.ror.tools.core.utils.RorToolsError.*
+import tech.beshu.ror.tools.core.utils.{EsDirectory, InOut, RorToolsError}
+
+import scala.util.{Failure, Success, Try}
+
+final class EsPatchExecutor(rorPluginDirectory: RorPluginDirectory,
+                            esPatch: EsPatch)
+                           (implicit inOut: InOut) {
+
+  def patch(): Either[RorToolsError, Unit] = {
+    checkWithPatchedByFileAndEsPatch() match {
+      case EsPatchStatus.PatchedWithCurrentRorVersion(rorVersion) =>
```

Comment:
Let's do `import EsPatchStatus.*`.IMO The errors are clear without adding this context explicitly.